ext/http/websocket_upgrade.rs

```diff
@@ -1,362 +0,0 @@
-  #[class("Http")]
-  #[error("attempted to write to completed upgrade buffer")]
-  UpgradeBufferAlreadyCompleted,
-}
-
-/// Given a buffer that ends in `\n\n` or `\r\n\r\n`, returns a parsed [`Request<Body>`].
-fn parse_response<T: Default>(
-  header_bytes: &[u8],
-) -> Result<(usize, Response<T>), WebSocketUpgradeError> {
-  let mut headers = [httparse::EMPTY_HEADER; 16];
-  let status = httparse::parse_headers(header_bytes, &mut headers)?;
-  match status {
-    Status::Complete((index, parsed)) => {
-      let mut resp = Response::builder().status(101).body(T::default())?;
-      for header in parsed.iter() {
-        resp.headers_mut().append(
-          HeaderName::from_bytes(header.name.as_bytes())?,
-          HeaderValue::from_str(std::str::from_utf8(header.value)?)?,
-        );
-      }
-      Ok((index, resp))
-    }
-    _ => Err(WebSocketUpgradeError::InvalidHeaders),
-  }
-}
-
-/// Find a newline in a slice.
-fn find_newline(slice: &[u8]) -> Option<usize> {
-  for (i, byte) in slice.iter().enumerate() {
-    if *byte == b'\n' {
-      return Some(i);
-    }
-  }
-  None
-}
-
-/// WebSocket upgrade state machine states.
-#[derive(Default)]
-enum WebSocketUpgradeState {
-  #[default]
-  Initial,
-  StatusLine,
-  Headers,
-  Complete,
-}
-
-static HEADER_SEARCHER: OnceCell<memchr::memmem::Finder> = OnceCell::new();
-static HEADER_SEARCHER2: OnceCell<memchr::memmem::Finder> = OnceCell::new();
-
-#[derive(Default)]
-pub struct WebSocketUpgrade<T: Default> {
-  state: WebSocketUpgradeState,
-  buf: BytesMut,
-  _t: PhantomData<T>,
-}
-
-impl<T: Default> WebSocketUpgrade<T> {
-  /// Ensures that the status line starts with "HTTP/1.1 101 " which matches all of the node.js
-  /// WebSocket libraries that are known. We don't care about the trailing status text.
-  fn validate_status(
-    &self,
-    status: &[u8],
-  ) -> Result<(), WebSocketUpgradeError> {
-    if status.starts_with(b"HTTP/1.1 101 ") {
-      Ok(())
-    } else {
-      Err(WebSocketUpgradeError::InvalidHttpStatusLine)
-    }
-  }
-
-  /// Writes bytes to our upgrade buffer, returning [`Ok(None)`] if we need to keep feeding it data,
-  /// [`Ok(Some(Response))`] if we got a valid upgrade header, or [`Err`] if something went badly.
-  pub fn write(
-    &mut self,
-    bytes: &[u8],
-  ) -> Result<Option<(Response<T>, Bytes)>, WebSocketUpgradeError> {
-    use WebSocketUpgradeState::*;
-
-    match self.state {
-      Initial => {
-        if let Some(index) = find_newline(bytes) {
-          let (status, rest) = bytes.split_at(index + 1);
-          self.validate_status(status)?;
-
-          // Fast path for the most common node.js WebSocket libraries that use \r\n as the
-          // separator between header lines and send the whole response in one packet.
-          if rest.ends_with(b"\r\n\r\n") {
-            let (index, response) = parse_response(rest)?;
-            if index == rest.len() {
-              return Ok(Some((response, Bytes::default())));
```

Comment:
This is probably OK to remove. Not sure how this fast path was used since it went through `httparse` anyway.